### PR TITLE
Support direct polymer escrow

### PR DESCRIPTION
**Pull request by reednaa**

### CODE REVIEW

File: src/collector/polymer/polymer.worker.ts [hunk 1]
```diff
@@ -50,7 +50,10 @@ class PolymerCollectorSnifferWorker {
         this.incentivesAddress = this.config.incentivesAddress;
         this.polymerAddress = this.config.polymerAddress;
         this.ibcEventEmitterInterface = IbcEventEmitter__factory.createInterface();
-        this.filterTopics = [[this.ibcEventEmitterInterface.getEvent('SendPacket').topicHash]];
+        this.filterTopics = [
```
Comment (jsanmigimeno) [inline src/collector/polymer/polymer.worker.ts:53]: I don't think this is correct. My understanding is that this means to get the events whose topic 0 is *either* 'SendPacket' or 'this.incentivesAddress'. To get the events whose topic 0 is 'SendPacket *and* topic 1 is 'this.incentivesAddress' it should be [[...SendPacket..., ...this.incentivesAddress...]]

File: src/collector/polymer/polymer.worker.ts [hunk 2]
```diff
@@ -246,43 +249,20 @@ class PolymerCollectorSnifferWorker {
     ): Promise<void> {
 
         const event = parsedLog.args as unknown as SendPacketEvent.OutputObject;
-        const destinationChain: string = event.sourceChannelId;
+
+        // We need to convert the source chain id to the destination chain using the
```
Comment (jsanmigimeno) [inline src/collector/polymer/polymer.worker.ts]:
```suggestion
        // We need to convert the source channel id to the destination chain using the
```

File: src/collector/polymer/polymer.worker.ts [hunk 3]
```diff
@@ -246,43 +249,20 @@ class PolymerCollectorSnifferWorker {
...
+        // settings map.
+        this.config.polymerChannels[event.sourceChannelId];
```
Comment (jsanmigimeno) [inline src/collector/polymer/polymer.worker.ts]: Remove this line
```suggestion
```

File: src/collector/polymer/polymer.worker.ts [hunk 4]
```diff
@@ -246,43 +249,20 @@ class PolymerCollectorSnifferWorker {
...
+        const destinationChain: string = this.config.polymerChannels[event.sourceChannelId] ?? event.sourceChannelId;
```
Comment (jsanmigimeno) [inline src/collector/polymer/polymer.worker.ts]: If the mapping of `sourceChannelId` within `polymerChannels` is not found I'd simply return early and not continue to process the event. Having an 'unknown' channel id as the `destinationChain` might cause problems further along the line.

**Review by jsanmigimeno — CHANGES_REQUESTED** [review]

File: src/collector/polymer/polymer.worker.ts (hunk 1 above)
Comment (reednaa) [reply src/collector/polymer/polymer.worker.ts:53]: You are correct. I think I was filtering on topic 0 is SendPacket or this.incentivesAddress and not caring what topic 1 was.

**Review by jsanmigimeno — APPROVED** [review]
👍

**Pull request merged**